### PR TITLE
BACKLOG-14321 integrate automatically orderable

### PR DESCRIPTION
**Pull request by Taybou**
## JIRA

https://jira.jahia.org/browse/BACKLOG-14321

## Tests

The following are included in this PR
- [x] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/FormBuilder.jsx [hunk 1]
```diff
@@ -19,7 +19,12 @@ const FormBuilder = ({mode}) => {
         return <></>;
     }
 
-    const isOrderingSection = !nodeData.isSite && !nodeData.isPage && nodeData.primaryNodeType.hasOrderableChildNodes && mode === Constants.routes.baseEditRoute;
+    const isAutomaticallyOrderable = nodeData.primaryNodeType.supertypes.find(s => s.name === Constants.automaticOrdering.mixinAutomaticallyOrderable);
```
Comment (jkevan) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/FormBuilder.jsx:22]: Why not reuse the same test: 
**const canAutomaticOrder = values && values[Constants.automaticOrdering.mixin] !== undefined;**
done in ChildrenSectionCmp ?
There is no need for specific check on jmix:automaticallyOrderable, jmix:orderedList is the extends that provide the fieldset, and it's extending jmix:automaticallyOrderable.
So the check **values && values[Constants.automaticOrdering.mixin] !== undefined;** should be enough.

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/ChildrenSection.jsx [hunk 2]
```diff
@@ -53,6 +54,8 @@ const styles = theme => ({
 export const ChildrenSectionCmp = ({section, classes, formik: {values, handleChange}}) => {
     const {t} = useTranslation();
     const {sections} = useContentEditorSectionContext();
+    const {nodeData} = useContentEditorContext();
+
     const canAutomaticOrder = values && values[Constants.automaticOrdering.mixin] !== undefined;
```
Comment (jkevan) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/ChildrenSection.jsx:59]: I would move this test in FormBuilder.jsx
and pass the canAutomaticOrder as prop to the ChildrenSection.
Also we could create and other prop to have:

```
const canAutomaticOrder = values && values[Constants.automaticOrdering.mixin] !== undefined;
const canManualOrder = nodeData.primaryNodeType.hasOrderableChildNodes;
```

and pass this to the ChildrenSection.jsx as props, like:

`<ChildrenSection key={section.displayName} section={section} canAutomaticOrder={canAutomaticOrder} canManualOrder={canManualOrder}/>`

File: src/javascript/ContentEditor.constants.js [hunk 3]
```diff
@@ -68,7 +68,8 @@ export const Constants = {
     },
     automaticOrdering: {
         section: 'listOrdering',
-        mixin: 'jmix:orderedList'
+        mixin: 'jmix:orderedList',
+        mixinAutomaticallyOrderable: 'jmix:automaticallyOrderable'
```
Comment (jkevan) [inline src/javascript/ContentEditor.constants.js:72]: We should continue to test on jmix:orderedList and not jmix:automaticallyOrderable,
so we could remove this jmix:automaticallyOrderable constant

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/FormBuilder.jsx [hunk 4]
```diff
@@ -19,7 +19,12 @@ const FormBuilder = ({mode}) => {
...
+    const isOrderingSection = !nodeData.isSite &&
+        !nodeData.isPage &&
+        (nodeData.primaryNodeType.hasOrderableChildNodes || isAutomaticallyOrderable) &&
```
Comment (jkevan) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/FormBuilder.jsx:25]: Replace with can **canAutomaticOrder || canManualOrder**

**Comment by Taybou** [comment]
@jkevan lot of ideas, I prefer discuss it with you when you’re available and do live test

**Pull request merged**